__tests__/api/BundlePush/BundlePusher.test.ts

```diff
@@ -0,0 +1,515 @@
+                                   "No bundle manifest file found:");
+    });
+    it("should complain with missing zOSMF profile for push", async () => {
+        const zosMFSpy = jest.spyOn(ZosmfSession, "createBasicZosmfSession").mockImplementationOnce(() => { throw new Error( "Injected zOSMF Create error" ); });
+        await runPushTestWithError("__tests__/__resources__/ExampleBundle01",  false, "Injected zOSMF Create error");
+
+        expect(zosMFSpy).toHaveBeenCalledTimes(1);
+    });
+    it("should complain with missing SSH profile for push", async () => {
+        const zosMFSpy = jest.spyOn(ZosmfSession, "createBasicZosmfSession").mockImplementationOnce(() => ({}));
+        const sshSpy = jest.spyOn(SshSession, "createBasicSshSession").mockImplementationOnce(() => { throw new Error( "Injected SSH Create error" ); });
+        await runPushTestWithError("__tests__/__resources__/ExampleBundle01",  false, "Injected SSH Create error");
+
+        expect(zosMFSpy).toHaveBeenCalledTimes(1);
+        expect(sshSpy).toHaveBeenCalledTimes(1);
+    });
+    it("should complain if remote bundle dir mkdir fails", async () => {
+        const zosMFSpy = jest.spyOn(ZosmfSession, "createBasicZosmfSession").mockImplementationOnce(() => ({}));
+        const sshSpy = jest.spyOn(SshSession, "createBasicSshSession").mockImplementationOnce(() => ({}));
+        const shellSpy = jest.spyOn(Shell, "executeSshCwd").mockImplementationOnce(() =>
+                                              { throw new Error( "Injected Shell error" ); });
+        await runPushTestWithError("__tests__/__resources__/ExampleBundle01",  false,
+              "A problem occurred attempting to run 'mkdir 12345678' in remote directory '/u/ThisDoesNotExist'. Problem is: Injected Shell error");
+
+        expect(zosMFSpy).toHaveBeenCalledTimes(1);
+        expect(sshSpy).toHaveBeenCalledTimes(1);
+        expect(shellSpy).toHaveBeenCalledTimes(1);
+    });
+    it("should complain if remote bundle dir error", async () => {
+        const zosMFSpy = jest.spyOn(ZosmfSession, "createBasicZosmfSession").mockImplementationOnce(() => ({}));
+        const sshSpy = jest.spyOn(SshSession, "createBasicSshSession").mockImplementationOnce(() => ({}));
+        const shellSpy = jest.spyOn(Shell, "executeSshCwd").mockImplementationOnce(() => ({}));
+        const listSpy = jest.spyOn(List, "fileList").mockImplementationOnce(() => { throw new Error( "Injected List error" ); });
+        await runPushTestWithError("__tests__/__resources__/ExampleBundle01",  false, "Injected List error");
+
+        expect(zosMFSpy).toHaveBeenCalledTimes(1);
+        expect(sshSpy).toHaveBeenCalledTimes(1);
+        expect(shellSpy).toHaveBeenCalledTimes(1);
+        expect(listSpy).toHaveBeenCalledTimes(1);
+    });
+    it("should complain if remote bundledir list fails", async () => {
+        const zosMFSpy = jest.spyOn(ZosmfSession, "createBasicZosmfSession").mockImplementationOnce(() => ({}));
```

Comment:
Could some of these spies that are common to most of the tests be set up in the beforeEach()   Would just make individual tests a bit easier to read.

Reply:
I've restructured the tests.

Reply:
Thanks, they're much clearer now.